### PR TITLE
solve android crash with CCLabelTTF setString function

### PR DESCRIPTION
**Pull request by jxgzxh111**

### CODE REVIEW

**Comment by waiter** [comment]
It's really a big bug!!!!! Please merge it.

File: cocos2dx/platform/android/java/src/org/cocos2dx/lib/Cocos2dxBitmap.java [hunk 1]
```diff
@@ -390,8 +390,11 @@ private static int computeY(final FontMetricsInt pFontMetricsInt,
 				}
 
 				/* Remove spaces at the beginning of a new line. */
-				while (pString.charAt(i) == ' ') {
-					++i;
+				if(i < charLength)
+				{
+					while (pString.charAt(i) == ' ') {
+						++i;
+					}
```
Comment (waiter) [inline cocos2dx/platform/android/java/src/org/cocos2dx/lib/Cocos2dxBitmap.java:397]: while (i < charLength && pString.charAt(i) == ' ') {
    ++i;
}

I think it must be like this.
@jxgzxh111

**Comment by minggo** [comment]
@jxgzxh111 I think @waiter 's suggestion is right.

**Comment by dumganhar** [comment]
Merged at https://github.com/cocos2d/cocos2d-x/pull/2907.
Thanks.

**Comment by yihuang** [comment]
What is this bug, how to reproduce?
My game have some crash reports, i'm not sure if they are due to this bug.

**Pull request closed without merging**